backend/src/main/java/net/es/oscars/web/rest/MacInfoController.java

```diff
@@ -0,0 +1,108 @@
+package net.es.oscars.web.rest;
+
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import net.es.oscars.web.beans.MacInfoRequest;
+import net.es.oscars.web.beans.MacInfoResponse;
+import net.es.oscars.sb.nso.rest.MacInfoResult;
+import net.es.oscars.sb.nso.rest.MacInfoServiceResult;
+import net.es.oscars.resv.svc.ConnService;
+import net.es.oscars.resv.ent.Connection;
+import net.es.oscars.resv.ent.VlanFixture;
+import net.es.oscars.sb.nso.LiveStatusFdbCacheManager;
+import net.es.oscars.sb.nso.db.NsoVcIdDAO;
+import net.es.oscars.sb.nso.ent.NsoVcId;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+public class MacInfoController {
+
+    @Autowired
+    private LiveStatusFdbCacheManager fdbCacheManager;
+
+    @Autowired
+    private NsoVcIdDAO nsoVcIdDAO;
+
+    @Autowired
+    private ConnService connSvc;
+
+    @RequestMapping(value = "/api/mac/info", method = RequestMethod.POST)
+    @ResponseBody
+    @Transactional
+    public MacInfoResponse getMacInfo(@RequestBody MacInfoRequest request) {
+        log.debug("MAC info request");
+        log.debug("Request:" + request.toString());
+
+        String connectionId = request.getConnectionId();
+        if(connectionId == null) {
+            log.info("MAC info request has no connection id!");
+            return null;
```

Comment:
Please throw exceptions when validating user arguments. That makes the HTTP response get a 4xx error code which, on the client side, is easier to deal with than checking for a null response.

I usually throw `NoSuchElementException` when they ask for something that doesn't exist, `IllegalArgumentException` when they pass in some bad value, etc. Or you can make your own exception class.

Reply:
okay, sounds good